lib/User/UserEntry.php

```diff
@@ -365,6 +365,20 @@ private function getAttributeValue($attributeName, $default = null, $trim = true
 			if ($trim) {
 				$value = \trim($value);
 			}
+
+			if ($attributeName === 'objectguid' || $attributeName === 'guid') {
+				$guidLength = \strlen($value);
+				// Sometimes we might get a binary guid (16 byte), so we decode it
+				if ($guidLength === 16) {
+					$value = Access::binGUID2str($value);
+				}
+
+				// GUID should be 16 bytes (binary) or 32 bytes decoded to string
+				if (($guidLength !== 32) && ($guidLength !== 16)) {
```

Comment:
It should be 36 not 32; we need to count the "-" chars too.

Reply:
:+1: